lib/mpt/NibbleSlice.sol

```diff
@@ -20,18 +24,75 @@ library NibbleSliceOps {
         return len(self) == 0;
     }
 
-    function eq(NibbleSlice memory self, NibbleSlice memory other) public pure returns (bool) {
+    function eq(NibbleSlice calldata self, NibbleSlice calldata other) public pure returns (bool) {
```

Comment:
Why calldata though?

Reply:
calldata allows to obtain bytes slices, this is required for the common prefix computation.

Reply:
Unfortunately the keys won't be calldata, they'll be constructed in memory. Let's just copy the arrays.

Reply:
No, not possibe. I'd implement a workaround with inline assembly, as solidity only support bytes slices for calldata types as of now